api/controllers/zetter.ts

```diff
@@ -1,294 +1,328 @@
-import tweets from '../data/tweets';
-import users from '../data/users';
-import followers from '../data/followers';
-import favorities from '../data/favorities';
 import { sign } from 'jsonwebtoken';
 import { createHash } from 'crypto';
 import { Request, Response } from 'express';
-import { FavoriteType, FollowerType, ProfileType, TweetType } from '../utils/types';
+import { prisma } from '../utils/prisma';
+import { Tweet, User, Favorite } from '@prisma/client';
 
-export const getTweets = (req: Request, res: Response) => {
+export const getTweets = async (req: Request, res: Response) => {
     const limit: number = 10;
-    const followingUsers: Array<FollowerType> = followers.filter((follower) => follower.from === req.user.id);
-    const followingUsersAndMyIds: Array<number> = followingUsers.map((obj) => obj.to);
-    let followingUsersTweetsList: Array<TweetType> = [];
-
-    followingUsersAndMyIds.push(req.user.id);
-    tweets.forEach((tweet) => {
-        if (followingUsersAndMyIds.includes(tweet.createdBy)) {
-            followingUsersTweetsList.push(tweet);
-        }
+    const user = await prisma.user.findUnique({
+        where: { id: req.user.id },
+        include: { following: true },
     });
+    if (!user) {
+        throw Error;
+    }
 
-    followingUsersTweetsList.sort(function (a: TweetType, b: TweetType) {
-        return Number(b.createdAt) - Number(a.createdAt);
-    });
+    const followingUsers = user.following;
+    const followingUsersAndMyIds: Array<number> = followingUsers.map((obj) => obj.id);
 
-    const result: Array<TweetType> = addInformationToTweet(followingUsersTweetsList, req);
-    res.json(result.slice(0, limit));
+    followingUsersAndMyIds.push(req.user.id);
+    const followingUsersTweetsList: Array<Tweet> = await prisma.tweet.findMany({
+        where: {
+            createdBy: {
+                in: followingUsersAndMyIds,
+            },
+        },
+        include: {
+            user: true,
+            favorities: true,
+        },
+        orderBy: {
+            createdAt: 'desc',
+        },
+        take: limit,
+    });
```

Comment:
こんな書き方もできる

- followingの`map`メソッドに渡すコールバックの引数は、`obj`のような名前ではなく、何を扱っているのか分かる名前（例えば `following`）にしたほうが可読性がぐっとあがる
- たんに要素を1つ追加するだけなら、配列を用意してpushしなくても、スプレッド構文で十分シンプルにかける（ `[...hoge, fuga]`, あるいは `[fuga, ...hoge]`）
- 細かいけど、自分のツイートも入っているという意味で、`followingUsersTweetsList`という名前はやめたほうがよい

```suggestion    
    const timelineTweetsList: Array<Tweet> = await prisma.tweet.findMany({
        where: {
            createdBy: {
                in: [user.id, ...user.following.map(following => following.id)],
            },
        },
        include: {
            user: true,
            favorities: true,
        },
        orderBy: {
            createdAt: 'desc',
        },
        take: limit,
    });
```